client/state/sites/selectors.js

```diff
@@ -68,36 +67,41 @@ export const getSiteBySlug = createSelector(
 );
 
 /**
- * Returns a normalized site object by its ID. Intends to replicate
- * the site object returned from the legacy `sites-list` module.
- *
+ * Returns a normalized site object by its ID or site slug.
  *
  * @param  {Object}  state  Global state tree
- * @param  {Number}  siteId Site ID
+ * @param  {Number}  siteId Site ID or site slug
  * @return {?Object}        Site object
  */
-export const getSite = createSelector(
-	( state, siteId ) => {
-		let site =
-			getRawSite( state, siteId ) ||
-			// Support for non-ID site retrieval
-			// Replaces SitesList#getSite
-			getSiteBySlug( state, siteId );
-
-		if ( ! site ) {
-			return null;
-		}
+let getSiteCache = new WeakMap();
 
-		// To avoid mutating the original site object, create a shallow clone
-		// before assigning computed properties
-		site = { ...site };
-		assign( site, getSiteComputedAttributes( state, siteId ) );
-		assign( site, getJetpackComputedAttributes( state, siteId ) );
+export function getSite( state, siteId ) {
+	const rawSite = getRawSite( state, siteId ) || getSiteBySlug( state, siteId );
```

Comment:
Did we still need the comment about support for non-id site retrieval? Or should we rename the param `siteId` to `siteIdOrSiteSlug`

Reply:
Renaming the param to `siteIdOrSlug`